lib/components/lock_widget.dart

```diff
@@ -50,7 +50,7 @@ class LockWidget extends StatelessWidget {
   }
 
   Widget getLockStateIcon(LockProvider lockProvider) {
-    return (lockProvider.loadingDetail || lockProvider.loadingAction)
+    return (lockProvider.isLoading || lockProvider.isPerformingAction)
```

Comment:
Could use lockProvider.isBusy instead of having to check for either isLoading or isPerformingAction